pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -982,6 +982,10 @@ func (a *Artifact) CleanPath() error {
 		return argoerrs.InternalErrorf("Artifact '%s' did not specify a path", a.Name)
 	}
 
+	// ensure path is separated by filepath.Separator (aka os.PathSeparator).
+	// This ensures e.g. on windows /foo/bar is translated to \foo\bar first - otherwise the regexps below wouldn't trigger.
+	path := filepath.FromSlash(a.Path)
```

Comment:
Yea that would definitely be a separate issue as I imagine there's more code it impacts than just this. Figured I'd mention it though.
The same logic can also be implemented function-level as well, though Go doesn't have native decorators to wrap that more cleanly and reusably.

> we'd need to convert it in a webhook

Do you mean in a handler func or middleware?

Reply:
I'm actually not sure if argo workflows has one or not, but I meant as a [validating admission webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) - or maybe mutating since we could automatically convert all paths to be able to treat them equally in the code base.

Reply:
Oh. I mean yes, technically most k8s operators are also admission controllers (I believe [`validate.go`](https://github.com/argoproj/argo-workflows/blob/08096fc0512ed57a89e4a95ced56512631d8c94b/workflow/validate/validate.go) is where most of that logic is). but I don't think that wouldn't be an appropriate place to handle this kind of logic -- it should be transparent to the user, while an admission controller is user-facing (both validation and mutation are user-facing).